src/pages/Scholarships.jsx

```diff
@@ -21,11 +24,60 @@ const useStyles = makeStyles((theme) => ({
 }));
 
 function ScholarshipsPage() {
+  const classes = useStyles();
   const [scholarships, setScholarships] = useState([]);
   const [loading, setLoading] = useState(true);
   const [error, setError] = useState();
-  const [sortField, setSortField] = useState('deadline');
-  const [sortDir, setSortDir] = useState('asc');
+  const history = useHistory();
+  const location = useLocation();
+  const params = queryString.parse(location.search);
+
+  const sortingOptions = {
+    deadlineSoon: {
+      field: 'deadline',
+      dir: 'asc',
+    },
+    deadlineLatest: {
+      field: 'deadline',
+      dir: 'desc',
+    },
+    amountLow: {
+      field: 'amount.min',
+      dir: 'asc',
+    },
+    amountHigh: {
+      field: 'amount.max',
+      dir: 'desc',
+    },
+  };
+
+  const validField =
+    params.sortField === 'deadline' || params.sortField === 'amount';
+  const validDir = params.sortDir === 'asc' || params.sortDir === 'desc';
+
+  const setSort = (val) => {
+    if (val in sortingOptions) {
+      history.push({
+        search: queryString.stringify({
+          sortField: sortingOptions[val].field,
+          sortDir: sortingOptions[val].dir,
+        }),
+      });
+    }
+  };
+
+  if (params.sortField === 'amount') {
+    if (validDir) {
+      if (params.sortDir === 'asc') {
+        setSort('amountLow');
+      } else {
+        setSort('amountHigh');
+      }
+    }
+  }
+
+  const sortField = validField ? params.sortField : DEFAULT_SORT_FIELD;
+  const sortDir = validDir ? params.sortDir : DEFAULT_SORT_DIR;
```

Comment:
I suggest defining `sortField` and `sortDir` before `validField` and `validDir` and instead [pruning](https://github.com/beyondhb1079/s4us/wiki/Guide:-Adding-URL-parameter-support#pruneparam) each invalid parameter so that by the time the code reaches this spot the sorting direction and field are valid.

Reply:
Wait what? Rn its using the validField to set them so how would I check it if its defined first

Reply:
I guess what I mean is this:

1. "define `sortField` and `sortDir` before (where) `validField` and `validDir` (are defined)":
```js
  const sortField = params.sortField ?? DEFAULT_SORT_FIELD;
  const sortDir = params.sortDir ?? DEFAULT_SORT_DIR;
```
2. Now what to do about `validField` and `validDir`? "[prune](https://github.com/beyondhb1079/s4us/wiki/Guide:-Adding-URL-parameter-support#pruneparam) each invalid parameter so that by the time the code reaches this spot the sorting direction and field are valid":
```js
  if (/* validField check fails */) {
    pruneQueryParam('sortField');
  }
  if (/* validDir check fails */) {
    pruneQueryParam('sortDir');
  }
  // Reaching here means sortDir and sortField are valid.
```

Reply:
Ok done